templates/vue/src/components/Lightbox/media/ActivityMedia/index.vue

```diff
@@ -113,40 +204,46 @@ export default {
   methods: {
     ...mapActions(["updateNodeProgress"]),
     markQuestionsComplete() {
-      for (let i = 0; i < this.questions.length; i++) {
-        const currentQuestion = this.questions[i]
-        const currentQuestionAnswer = this.getAnswers(
-          this.node.id,
-          currentQuestion.id
-        )
-        if (Object.keys(currentQuestionAnswer).length === 0) {
-          currentQuestion.completed = false
+      this.questions.forEach(question => {
+        const answer = this.getAnswers(this.questionNode.id, question.id)
+        if (Object.entries(answer).length === 0) {
+          question.completed = false
         } else {
-          currentQuestion.completed = true
-        }
-      }
-    },
-    handleSubmit() {
-      this.showCompletionScreen = true
-      const numberCompleted = this.questions.filter(question => question.completed)
-        .length
-      const progress = numberCompleted / this.node.typeData.activity.questions.length
-      this.updateNodeProgress({ id: this.node.id, progress }).then(() => {
-        if (progress === 1) {
-          this.$emit("complete")
+          question.completed = true
         }
       })
     },
+    handleComplete(sender) {
+      if (sender === states.ACTIVITY) {
```

Comment:
I find it a bit confusing to use the states for senders as well, even though they match in their string values.